compiler/circle-mpqsolver/src/core/TestHelper.test.cpp

```diff
@@ -208,5 +208,36 @@ bool isFileExists(const std::string &path)
 }
 
 } // namespace io_utils
+
+namespace data_utils
+{
+
+class SampleDataInput final : public core::DataProvider
+{
+public:
+  SampleDataInput() : _num_of_samples(1) {}
+  size_t numSamples() const override { return _num_of_samples; }
+  uint32_t numInputs(uint32_t sample) const override { return 1; }
+  void getSampleInput(uint32_t sample, uint32_t input, core::InputData &data) const
+  {
+    size_t size = data.data().size() / sizeof(float);
```

Comment:
Q) just curious and my be stupid question, if we can use number of elements in `core::InputData` are used, why do we pass `sample` and `input`?

Reply:
> Q) just curious and my be stupid question, if we can use number of elements in `core::InputData` are used, why do we pass `sample` and `input`?

@seanshpark
`Sample` stands for specific `data_point`, this `data_point` in its turn can have any number of tensors, so it's two-dimensional, that is why we need two parameters to query specific data. `size` it's just a number of elements in `data`. These parameters are unrelated. So we need them all. Quantity of inputs for specific `sample` can be queried by `numInputs`.